### PR TITLE
Relax protobuf minimum version dependency to 3.0.0

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The minimum version seems to be pretty arbitrarily picked so might as
well go back a bit further to match that of `ddtrace`.

### CODE REVIEW

**Review by brettlangdon — APPROVED** [review]
any particular reason to do 3.6.0?

any blockers for supporting >= 3.0 ?

**Comment by Kyle-Verhoog** [comment]
Yeah there's a bit more context here as the PR evolved as I tried going as far back as possible, I'll update the PR description.

**Comment by Kyle-Verhoog** [comment]
I was able to get 3.0 working quite easily thanks to @jd so we might as well just match `ddtrace`'s version.

File: releasenotes/notes/protobuf-min-f6af9a2d5d96f53c.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+features:
+  - |
+    Add support for protobuf>=3.0.0.
```
Comment (brettlangdon) [inline releasenotes/notes/protobuf-min-f6af9a2d5d96f53c.yaml:4]: "Other"?

As a feature this would imply a minor version bump?

Comment (Kyle-Verhoog) [reply releasenotes/notes/protobuf-min-f6af9a2d5d96f53c.yaml:4]: Hmm right. Yeah let's do it as other.

**Review by brettlangdon — APPROVED** [review]

**Review by jd — APPROVED** [review]

**Pull request merged**